cmd/query/app/querysvc/query_service_v2.go

```diff
@@ -0,0 +1,141 @@
+// // Copyright (c) 2019 The Jaeger Authors.
+// // SPDX-License-Identifier: Apache-2.0
+
+package querysvc
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/model/adjuster"
+	"github.com/jaegertracing/jaeger/storage/spanstore"
+	"github.com/jaegertracing/jaeger/storage_v2/depstore"
+	"github.com/jaegertracing/jaeger/storage_v2/tracestore"
+)
+
+// TODO: Remove query_service.go and rename query_service_v2.go
+// to query_service.go once all components have been migrated to
+// operate on the OTEL data model.
+var errNoArchiveSpanStorageV2 = errors.New("archive span storage was not configured")
+
+const (
+	defaultMaxClockSkewAdjustV2 = time.Second
+)
+
+// QueryServiceOptions has optional members of QueryService
+type QueryServiceOptionsV2 struct {
+	ArchiveTraceReader tracestore.Reader
+	ArchiveTraceWriter tracestore.Writer
+	Adjuster           adjuster.Adjuster
+}
+
+// StorageCapabilities is a feature flag for query service
+type StorageCapabilitiesV2 struct {
+	ArchiveStorage bool `json:"archiveStorage"`
+	// TODO: Maybe add metrics Storage here
+	// SupportRegex     bool
+	// SupportTagFilter bool
+}
+
+// QueryService contains span utils required by the query-service.
+type QueryServiceV2 struct {
+	traceReader      tracestore.Reader
+	dependencyReader depstore.Reader
+	options          QueryServiceOptionsV2
+}
+
+// NewQueryService returns a new QueryService.
+func NewQueryServiceV2(traceReader tracestore.Reader, dependencyReader depstore.Reader, options QueryServiceOptions) *QueryService {
+	qsvc := &QueryService{
+		traceReader:      traceReader,
+		dependencyReader: dependencyReader,
+		options:          options,
+	}
+
+	if qsvc.options.Adjuster == nil {
+		qsvc.options.Adjuster = adjuster.Sequence(StandardAdjusters(defaultMaxClockSkewAdjustV2)...)
```

Comment:
did we already implement these? The need to operate on OTLP.

Reply:
Great project for LLM

Reply:
@yurishkuro I'll do this first in a separate PR. Currently these live in `model/adjuster`. Which package should hold the new interface / components?

Reply:
`cmd/query/app/internal/api_v3/adjuster` might be ok

Reply:
actually, `cmd/query/app/querysvc/adjuster` is better, given why it was exposed in the first place (Uber had additional internal adjusters)

Reply:
Sounds good - I booked #6344 to track this effort